test/pytest/receiver/test_asapo_transfer.py

```diff
@@ -0,0 +1,277 @@
+                 'commissioning/raw/foo1/bar1_1.h5', 'commissioning/raw/foo1/bar2_1.h5',
+                 'commissioning/raw/foo2/bar1_1.h5', 'commissioning/raw/foo2/bar2_1.h5',
+                 ]
+    return file_list
+
+
+@pytest.fixture
+def hidra_metadata(file_list):
+    hidra_metadata = []
+    for file_path in file_list:
+        hidra_metadata.append({'source_path': 'http://127.0.0.1/data', 'relative_path': path.dirname(file_path),
+                               'filename': path.basename(file_path), 'version': '4.4.2', 'chunksize': 10485760,
+                               'file_mod_time': 1643637926.004875,
+                               'file_create_time': 1643637926.0048773, 'confirmation_required': False,
+                               'chunk_number': 0})
+    return hidra_metadata
+
+
+@pytest.fixture
+def worker(config):
+    worker_config = config.copy()
+    del worker_config["user_config_path"]
+    worker = AsapoWorker(**worker_config)
+    worker.send_message = create_autospec(worker.send_message)
+    yield worker
+
+
+@pytest.fixture
+def query(config):
+    query = create_autospec(Transfer, instance=True)
+    yield query
+
+
+@pytest.fixture
+def asapo_transfer(worker, query, transfer_config, hidra_metadata):
+    asapo_transfer = AsapoTransfer(worker, query,
+                                   transfer_config['target_host'],
+                                   transfer_config['target_port'],
+                                   transfer_config['target_dir'], transfer_config['reconnect_timeout'])
+    return_list = [[metadata, None] for metadata in hidra_metadata]
+    asapo_transfer.query.get.side_effect = return_list
+    yield asapo_transfer
+
+
+def test_varify_config(config):
+    with pytest.raises(ValueError):
+        varify_config(config)
+    del config["user_config_path"]
+    del config["beamline"]
+    del config["token"]
+    varify_config(config)
+
+
+def test_asapo_transfer(file_list, asapo_transfer, hidra_metadata):
+
+    x = threading.Thread(target=asapo_transfer.run, args=())
+    x.start()
+    sleep(1)
+    asapo_transfer.stop()
+    sleep(1)
+    x.join()
+    asapo_transfer.asapo_worker.send_message.assert_called_with(file_list[-1], hidra_metadata[-1])
+    asapo_transfer.query.stop.assert_called_with()
+
+
+def create_mock_transfer(asapo_worker, *kwargs):
+    asapo_transfer = Mock()
+    asapo_transfer.run = Mock(return_value=True)
+    asapo_worker.send_message(None, None)
+    return asapo_transfer
+
+
+def create_mock_failing_transfer(asapo_worker, *kwargs):
+    asapo_transfer = Mock()
+    asapo_transfer.run = Mock()
+    asapo_transfer.run.side_effect = Exception('mocked error')
+    asapo_worker.send_message(None, None)
+    return asapo_transfer
+
+
+@patch('asapo_transfer.create_query', create_autospec(create_query, instance=True))
+@patch('asapo_transfer.create_asapo_transfer', create_mock_transfer)
+def test_run_transfer(worker, transfer_config_obj, file_list, hidra_metadata):
```

Comment:
What issues does this test is supposed to find? It takes a lot of time but because of all the mocks it is not clear if it tests anything important?